screens/modals/QuantityModal.js

```diff
@@ -0,0 +1,162 @@
+import { FontAwesome5 } from '@expo/vector-icons';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { Modal, TouchableOpacity, View } from 'react-native';
+import Colors from '../../assets/Colors';
+import { Body, ButtonLabel, Title } from '../../components/BaseComponents';
+import LineItemCard from '../../components/LineItemCard';
+import ProductDisplayCard from '../../components/ProductDisplayCard';
+import {
+  ModalCenteredOpacityLayer,
+  ModalContentContainer,
+  ModalCopyContainer,
+  QuantityInput
+} from '../../styled/modal';
+import { ColumnContainer, RoundedButtonContainer } from '../../styled/shared';
+
+export default class QuantityModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalVisible: false,
+      product: null,
+      // Needs to be string type because it's TextInput
+      currentQuantity: '',
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    const { product } = this.props;
+    this.resetCurrentQuantity();
+    this.setState({
+      product,
+      isLoading: false
+    });
+  }
+
+  // Forces a re-render when new props are passed
+  componentWillReceiveProps(nextProps) {
```

Comment:
Interesting, did not know this function was a thing! What’s the difference between this and `componentDidUpdate` do you know?

Reply:
Apparently the `willReceiveProps` is deprecated LOL

Reply:
"I won't look at the comments until later" chronically lying about not .. coding. Damn I didn't realize 😂 https://blog.bitsrc.io/react-16-lifecycle-methods-how-and-when-to-use-them-f4ad31fb2282 I looked at the old version of this blog post LOL -- anyway, I think it's been replaced by `getDerivedStateFromProps` which is.. an incredibly specific name and exactly what I'm doing.

Also, the fact that the author does not seem to like this function 😂
> Even with the above cases, there’s usually a better way to do it. But getDerivedStateFromProps will have your back when worst comes to worst.

 I think `componentDidUpdate` doesn't seem to be the same use case? So I'm going to go ahead and replace it with `getDerivedStateFromProps`  ..

I shall change this at minimum in the rewards redesign PR, which apparently will become an updating modals PR (it's tentatively done; maybe I'll make a draft PR after finishing HW https://github.com/calblueprint/dccentralkitchen-clerks/compare/annie/quantity-modal...annie/reward-redesign)